clients/src/main/java/org/apache/kafka/common/requests/UpdateMetadataRequest.java

```diff
@@ -74,6 +75,43 @@ public String toString() {
         }
     }
 
+    // TODO: rename this class to UpdateMetadataPartitionState and probably move it to its own class
```

Comment:
I think the current class name UpdateMetadataRequestPartitionState is good enough. I have removed this TODO.